packages/kbn-securitysolution-exception-list-components/src/exception_item_card/conditions/entry_content/entry_content.helper.tsx

```diff
@@ -21,13 +21,21 @@ const entryValueWrapStyle = css`
 const EntryValueWrap = ({ children }: { children: React.ReactNode }) => (
   <span className={entryValueWrapStyle}>{children}</span>
 );
-const getEntryValue = (type: string, value?: string | string[]) => {
+
+const getEntryValue = (type: string, value: string | string[], showValueListModal: ElementType) => {
```

Comment:
Same thing here: I don't think this needs to be an argument to `getEntryValue`, you can just add the new logic branch with an explicit `ShowValueListModal` reference 👍 .

Reply:
I totally agree and I don't like this props drilling myself.

The problem that those Exceptiond and Autocomplete are - packages - and only way I found it's to pass this modal window as props.

We already do that, and have those comments

```
securityLinkAnchorComponent: ElementType; // This property needs to be removed to avoid the Prop Drilling, once we move all the common components from x-pack/security-solution/common
```

Alternative solutions - can be to move this modal window to another package, and make as dependency for Exception and Autocomplete - But I don't see why a very specific business logic component should go to another package. 

Please let me know if you have other ideas

Reply:
So this is what I'm thinking:

- ++ what @rylnd suggested of moving the value lists components out to avoid passing through the modal as a prop
- I do wish we could have the modal not be nested so deeply within another modal (in the exceptions modal). We've observed bugs of unintended re-renders of the exception conditions and that could man we see the value list modal begin to unexpectedly re-render as well. Could we run this through the flakey test runner to see if we catch any of that behavior?
- I would love to chat tomorrow during tech time to understand what options we have here. I think what Ryland suggested could be a step further. The exceptions/lists workflows are under review, so this is also a great time to discuss design patterns we'd like to move towards. We can certainly take an iterative approach here.

Reply:
After discussion today, we decided that we will not move Value list modal to another packages, as it already depends on a lot of things from security-solution plugin.

Otherwhise there was idea to move Exceptions component package back into security solution, which I will create issue for.

Reply:
> I do wish we could have the modal not be nested so deeply within another modal 

Agree, but don't know what other options we can have here. Can be expanded flyout, but exceptions flyout already to wide, it will be not enough place for that.

Reply:
>Could we run this through the flakey test runner to see if we catch any of that behavior?

would run some flaky test runner